atlasdb-config/src/main/java/com/palantir/atlasdb/http/FailoverFeignTarget.java

```diff
@@ -150,10 +151,13 @@ private void checkAndHandleFailure(RetryableException ex) {
 
 
     private void pauseForBackoff(RetryableException ex) {
-        double pauseTime = Math.pow(
+        double exponentialPauseTime = Math.pow(
                 GOLDEN_RATIO,
                 numSwitches.get() * failuresBeforeSwitching + failuresSinceLastSwitch.get());
-        pauseForBackoff(ex, Math.round(pauseTime));
+        long cappedPauseTime = Math.min(maxBackoffMillis, Math.round(exponentialPauseTime));
+        long pauseTimeWithJitter = ThreadLocalRandom.current().nextLong(cappedPauseTime);
```

Comment:
should probably include comment that we're using `Full jitter` from https://www.awsarchitectureblog.com/2015/03/backoff.html for reader's awareness